Framework/Algorithms/src/CrossCorrelate.cpp

```diff
@@ -175,7 +176,10 @@ void CrossCorrelate::exec() {
     size_t wsIndex = indexes[i]; // Get the ws index from the table
     // Copy spectra info from input Workspace
     out->getSpectrum(i).copyInfoFrom(inputWS->getSpectrum(wsIndex));
-    out->mutableX(i) = XX;
+
+    if (i > 0) {
+      out->setSharedX(i, out->sharedX(0));
+    }
```

Comment:
Note: below is another good example for histogram arithmetics, please add to your list.